src/main/java/xyz/nucleoid/plasmid/command/MapCommand.java

```diff
@@ -193,6 +224,74 @@ private static int addRegion(CommandContext<ServerCommandSource> context) throws
         return Command.SINGLE_SUCCESS;
     }
 
+    private static int renameRegions(CommandContext<ServerCommandSource> context, BlockStateVariantMap.TriFunction<TemplateRegion, String, BlockPos, Boolean> predicate) throws CommandSyntaxException {
+        ServerCommandSource source = context.getSource();
+        ServerWorld world = source.getWorld();
+        BlockPos pos = source.getPlayer().getBlockPos();
+
+        StagingMapManager stagingMapManager = StagingMapManager.get(world);
+
+        String oldMarker = StringArgumentType.getString(context, "old");
+        String newMarker = StringArgumentType.getString(context, "new");
+
+        Optional<StagingMapTemplate> mapOpt = stagingMapManager.getStagingMaps().stream()
```

Comment:
Would it be possible to extract this common code for finding staging map & templates?